src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -82,9 +84,14 @@ private static boolean isValidPath(Path filePath) {
      * Saves all data to this storage file.
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws FileNotFoundException if the file is not found before saving
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
+    	
+    	if (!checkFilePathExists() && fileIsInitFromStart) {
```

Comment:
Correct way to check the existence of the file. Also consider creating the file and show the problem to the user, instead of closing the program.